pepper-apis/src/main/java/org/broadinstitute/ddp/analytics/GoogleAnalyticsMetricsTracker.java

```diff
@@ -0,0 +1,121 @@
+        //if (noAnalyticsTokenStudies.contains(studyGuid) || getStudySettingByStudyGuid(studyGuid).isAnalyticsEnabled()) {
+        if (noAnalyticsTokenStudies.contains(studyGuid)) {
+            return;
+        }
+
+        GoogleAnalytics metricTracker = getMetricTracker(studyGuid);
+        if (metricTracker != null) {
+            metricTracker.event().eventCategory(eventHit.eventCategory())
+                    .eventAction(eventHit.eventAction())
+                    .eventLabel(eventHit.eventLabel())
+                    .eventValue(eventHit.eventValue())
+                    .sendAsync();
+        }
+    }
+
+    public static StudySettings getStudySettingByStudyGuid(String studyGuid) {
+        //todo: revisit after Redis caching to use cached Study, StudySettings and get rid of StudySettingsStore
+        return StudySettingsStore.getInstance().getStudySettings(studyGuid);
+    }
+
+
+    private static class StudySettingsStore {
+        private static StudySettingsStore instance;
+        private static Object lockVar1 = "lock";
+        private static Map<String, StudySettings> studySettings = new HashMap<>();
+
+        private static StudySettingsStore getInstance() {
+            if (instance == null) {
+                synchronized (lockVar1) {
+                    if (instance == null) {
+                        instance = new StudySettingsStore();
+                        loadAllStudySettings();
+                    }
+                }
+            }
+            return instance;
+        }
+
+        private static synchronized void loadAllStudySettings() {
+            studySettings = TransactionWrapper.withTxn(handle -> {
+                int studiesWithSettings = 0;
+                JdbiUmbrellaStudy jdbiUmbrellaStudy = handle.attach(JdbiUmbrellaStudy.class);
+                StudyDao studyDao = handle.attach(StudyDao.class);
+                List<StudyDto> studyDtos = jdbiUmbrellaStudy.findAll();
+                Map<String, StudySettings> settingsMap = new HashMap<>();
+                for (StudyDto studyDto : studyDtos) {
+                    //load StudySettings
+                    Optional<StudySettings> studySettings = studyDao.findSettings(studyDto.getId());
+                    if (studySettings.isPresent()) {
+                        settingsMap.put(studyDto.getGuid(), studySettings.get());
+                        studiesWithSettings++;
+                    } else {
+                        //no StudySettings for this study.. create dummy one to avoid unnecessary future DB calls
+                        settingsMap.put(studyDto.getGuid(), new StudySettings(0, null, false, null));
+                    }
+                }
+                LOG.info("Loaded StudySettings for {} studies.", studiesWithSettings);
+                return settingsMap;
+            });
+        }
+
+        private StudySettings getStudySettings(String studyGuid) {
```

Comment:
Can we make this return `Optional<StudySettings>` instead of creating dummy objects?
* Use `..map(StudySettings::getAnalyticsToken).orElse(null)` to grab the token,
* and `..orElse(null)` to grab the settings object itself.

Reply:
did some cleanup